processing/src/main/java/org/apache/carbondata/processing/newflow/encoding/impl/RowEncoderImpl.java

```diff
@@ -0,0 +1,132 @@
+import org.apache.carbondata.processing.newflow.CarbonDataLoadConfiguration;
+import org.apache.carbondata.processing.newflow.DataField;
+import org.apache.carbondata.processing.newflow.constants.DataLoadProcessorConstants;
+import org.apache.carbondata.processing.newflow.encoding.FieldEncoder;
+import org.apache.carbondata.processing.newflow.encoding.RowEncoder;
+import org.apache.carbondata.processing.newflow.exception.CarbonDataLoadingException;
+import org.apache.carbondata.processing.newflow.row.CarbonRow;
+import org.apache.carbondata.processing.util.RemoveDictionaryUtil;
+
+/**
+ *
+ */
+public class RowEncoderImpl implements RowEncoder {
+
+  private CarbonDataLoadConfiguration configuration;
+
+  private AbstractDictionaryFieldEncoderImpl[] dictionaryFieldEncoders;
+
+  private NonDictionaryFieldEncoderImpl[] nonDictionaryFieldEncoders;
+
+  private MeasureFieldEncoderImpl[] measureFieldEncoders;
+
+  public RowEncoderImpl(DataField[] fields, CarbonDataLoadConfiguration configuration) {
+    this.configuration = configuration;
+    CacheProvider cacheProvider = CacheProvider.getInstance();
+    Cache<DictionaryColumnUniqueIdentifier, Dictionary> cache = cacheProvider
+        .createCache(CacheType.REVERSE_DICTIONARY,
+            configuration.getTableIdentifier().getStorePath());
+    List<AbstractDictionaryFieldEncoderImpl> dictFieldEncoders = new ArrayList<>();
+    List<NonDictionaryFieldEncoderImpl> nonDictFieldEncoders = new ArrayList<>();
+    List<MeasureFieldEncoderImpl> measureFieldEncoderList = new ArrayList<>();
+
+    long lruCacheStartTime = System.currentTimeMillis();
+
+    for (int i = 0; i < fields.length; i++) {
+      FieldEncoder fieldEncoder = FieldEncoderFactory.getInstance()
+          .createFieldEncoder(fields[i], cache,
+              configuration.getTableIdentifier().getCarbonTableIdentifier(), i);
+      if (fieldEncoder instanceof AbstractDictionaryFieldEncoderImpl) {
+        dictFieldEncoders.add((AbstractDictionaryFieldEncoderImpl) fieldEncoder);
+      } else if (fieldEncoder instanceof NonDictionaryFieldEncoderImpl) {
+        nonDictFieldEncoders.add((NonDictionaryFieldEncoderImpl) fieldEncoder);
+      } else if (fieldEncoder instanceof MeasureFieldEncoderImpl) {
+        measureFieldEncoderList.add((MeasureFieldEncoderImpl)fieldEncoder);
+      }
+    }
+    CarbonTimeStatisticsFactory.getLoadStatisticsInstance()
+        .recordLruCacheLoadTime((System.currentTimeMillis() - lruCacheStartTime) / 1000.0);
+    dictionaryFieldEncoders =
+        dictFieldEncoders.toArray(new AbstractDictionaryFieldEncoderImpl[dictFieldEncoders.size()]);
+    nonDictionaryFieldEncoders = nonDictFieldEncoders
+        .toArray(new NonDictionaryFieldEncoderImpl[nonDictFieldEncoders.size()]);
+    measureFieldEncoders = measureFieldEncoderList
+        .toArray(new MeasureFieldEncoderImpl[measureFieldEncoderList.size()]);
+
+  }
+
+  @Override public CarbonRow encode(CarbonRow row) throws CarbonDataLoadingException {
```

Comment:
Ok. Here I will remove it and moved to sort step to keep sort preparation at one place.